### PR TITLE
Skip resize if fs is readonly

### PR DESCRIPTION
**Pull request by judemars**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

See https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/1088. Right now we attempt a filesystem resize without checking whether it's readonly. This change checks the readonly bit of the filesystem, and skips resizing if it is set to "1".

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/mount-utils/resizefs_linux.go [hunk 1]
```diff
@@ -159,6 +174,22 @@ func (resizefs *ResizeFs) getDeviceSize(devicePath string) (uint64, error) {
 	return size, nil
 }
 
+func (resizefs *ResizeFs) getDeviceRO(devicePath string) (bool, error) {
+	output, err := resizefs.exec.Command(blockDev, "--getro", devicePath).CombinedOutput()
+	outStr := strings.TrimSpace(string(output))
+	if err != nil {
+		return false, fmt.Errorf("failed to read size of device %s: %s: %s", devicePath, err, outStr)
```
Comment (leiyiz) [inline staging/src/k8s.io/mount-utils/resizefs_linux.go:181]: nit: should the log here be `failed to get read-only ....` or something similar?

**Comment by leiyiz** [comment]
/approve
please squash and will lgtm

**Comment by leiyiz** [comment]
ahh looks like I don't have approval power

**Comment by gnufied** [comment]
I will approve but although a nit, it would be good to have log message fixed.

File: staging/src/k8s.io/mount-utils/resizefs_linux.go (hunk 1 above)
Comment (judemars) [reply staging/src/k8s.io/mount-utils/resizefs_linux.go]: Woops, fixed. Thank you!

**Comment by judemars** [comment]
@leiyiz I set the merge method to squash, I think it's easier that way in case of last minute changes

**Pull request merged**